EdvClient.js

```diff
@@ -1016,7 +1016,14 @@ export class EdvClient {
 
     if(update) {
       if('sequence' in encrypted) {
-        encrypted.sequence++;
+        // Sequence is limited to MAX_SAFE_INTEGER - 1 to avoid unexpected
+        // behavior when a client attempts to increment the sequence number.
+        if(Number.isSafeInteger(encrypted.sequence) &&
+          encrypted.sequence < Number.MAX_SAFE_INTEGER - 1) {
+          encrypted.sequence++;
+        } else {
+          throw new Error('"sequence" is too large.');
+        }
```

Comment:
Stylistically, we try to avoid additional nesting when we can because it increases code readability (often shortens code and involves less indention). So this usually manifests by doing error checks first and throwing -- and then we don't need to nest thereafter. This also often helps get all of the validation code out of the way before any of the implementation code happens vs. having some validations checks at the top, then some nested implementation, then the results of failing validation via an `else`. All of that can help with readability as well.

In short, we can get that sort of thing by structuring the code this way:

```suggestion
        if(!Number.isSafeInteger(encrypted.sequence) ||
          encrypted.sequence >= Number.MAX_SAFE_INTEGER - 1) {
          throw new Error('"sequence" is too large.');
        }
        encrypted.sequence++;
```

Note that the full conditional could have just been negated as well when parenthesis if that's more logically readable (sometimes it is, sometimes it isn't):

```suggestion
        if(!(Number.isSafeInteger(encrypted.sequence) &&
          encrypted.sequence < Number.MAX_SAFE_INTEGER - 1)) {
          throw new Error('"sequence" is too large.');
        }
        encrypted.sequence++;
```

Either of those is ok, IMO, but the former seemed clearer to me in this case.

Reply:
Makes sense! I implemented the former code structure, but switched the second statement to !(encrypted.sequence < Number.MAX_SAFE_INTEGER - 1) because of the previous discussion that preferred using less than instead of greater than to compare MAX_SAFE_INTEGER.